src/high-level/types/complex-double-float.lisp

```diff
@@ -0,0 +1,272 @@
+;;;; complex-double-float.lisp
+;;;;
+;;;; Author: Cole Scott
+
+(in-package #:magicl)
+
+(deftensor tensor/complex-double-float (complex double-float))
+
+(defmatrix matrix/complex-double-float (complex double-float) tensor/complex-double-float)
+
+(defvector vector/complex-double-float (complex double-float) tensor/complex-double-float)
+
+(defcompatible
+    (lambda (tensor)
+      (case (order tensor)
+        (1 '(vector/complex-double-float
+             tensor/complex-double-float))
+        (2 '(matrix/complex-double-float
+             tensor/complex-double-float))
+        (t '(tensor/complex-double-float))))
+  tensor/complex-double-float
+  matrix/complex-double-float
+  vector/complex-double-float)
+
+(defmethod dot ((vector1 vector/complex-double-float) (vector2 vector/complex-double-float))
+  (assert (cl:= (size vector1) (size vector2))
+          () "Vectors must have the same size. The first vector is size ~a and the second vector is size ~a."
+          (size vector1) (size vector2))
+  (loop :for i :below (size vector1)
+        :sum (* (tref vector1 i) (conjugate (tref vector2 i)))))
+
+(defmethod orthonormalize! ((m matrix/complex-double-float))
+  "Applies Gram-Schmidt to the columns of a full rank square matrix to produce a unitary matrix, replacing the elements"
```

Comment:
```suggestion
  "Applies the Modified Gram-Schmidt algorithm to the columns of a full rank square matrix to produce a unitary matrix, replacing the elements"
```
Added mainly b/c "Gram-Schmidt" is a classic example of a numerically unstable algorithm, although the variant implemented here does not suffer from this (I think). Also FWIW it's reasonable to just drop this whole thing and use one of the QR routines.

Reply:
Yeah I'm going to go ahead and just drop these methods

Reply:
65f6de6